File: neoforge/src/main/java/net/typho/vibrancy/VibrancyNeoForge.java
package net.typho.vibrancy;

import net.minecraft.core.particles.ParticleType;
import net.minecraft.core.particles.SimpleParticleType;
import net.minecraft.core.registries.BuiltInRegistries;
import net.neoforged.fml.common.Mod;
import net.neoforged.neoforge.registries.DeferredRegister;

import java.util.function.Supplier;

@Mod(Vibrancy.MOD_ID)
public class VibrancyNeoForge {
    public static final DeferredRegister<ParticleType<?>> PARTICLES = DeferredRegister.create(BuiltInRegistries.PARTICLE_TYPE, Vibrancy.MOD_ID);
    public static final Supplier<SimpleParticleType> STEAM = PARTICLES.register(
            "steam",
            () -> new SimpleParticleType(false)
    );

    public VibrancyNeoForge() {
        Vibrancy.STEAM = STEAM;
        Vibrancy.init();
    }
}

File: neoforge/src/main/kotlin/net/typho/vibrancy/VibrancyNeoForge.kt
package net.typho.vibrancy

import net.neoforged.api.distmarker.Dist
import net.neoforged.bus.api.IEventBus
import net.neoforged.fml.ModContainer
import net.neoforged.fml.common.Mod
import net.neoforged.neoforge.client.gui.IConfigScreenFactory

@Mod(value = Vibrancy.MOD_ID, dist = [Dist.CLIENT])
class VibrancyNeoForge(eventBus: IEventBus, modContainer: ModContainer) {
    init {
        modContainer.registerExtensionPoint(IConfigScreenFactory::class.java, IConfigScreenFactory { container, modListScreen -> VibrancyConfig.createScreen(modListScreen) })
    }
}